ambry-frontend/src/test/java/com.github.ambry.frontend/AmbryBlobStorageServiceTest.java

```diff
@@ -399,6 +469,113 @@ public void postGetHeadDeleteTest() throws Exception {
+   * A#UNKNOWN        realCntName       "someServiceId"     InvalidAccount      null                  null
+   * realAcctName     null              "someServiceId"     MissingArgs         null                  null
+   * realAcctName     nonExistName      "someServiceId"     InvalidContainer    null                  null
+   * realAcctName     C#UNKOWN          "someServiceId"     InvalidContainer    null                  null
+   * realAcctName     realCntName       "someServiceId"     null                realAccount           realContainer
+   * nonExistName     null              "someServiceId"     MissingArgs         null                  null
+   * nonExistName     nonExistName      "someServiceId"     InvalidAccount      null                  null
+   * nonExistName     C#UNKOWN          "someServiceId"     InvalidAccount      null                  null
+   * nonExistName     realCntName       "someServiceId"     InvalidAccount      null                  null
+   * null             null              A#UNKNOWN           InvalidAccount      null                  null
+   * null             nonExistName      A#UNKNOWN           InvalidAccount      null                  null
+   * null             C#UNKOWN          A#UNKNOWN           InvalidAccount      null                  null
+   * null             realCntName       A#UNKNOWN           InvalidAccount      null                  null
+   * null             null              realAcctName        InvalidContainer    null                  null     Note: The account does not have the two default containers for legacy public and private blobs.
+   * null             nonExistName      realAcctName        MissingArgs         null                  null
+   * null             C#UNKOWN          realAcctName        InvalidContainer    null                  null
+   * null             realCntName       realAcctName        MissingArgs         null                  null
+   * null             null              realAcctName        null                realAccount           default pub/private ctn     Note: The account has the two default containers for legacy public and private blobs.
+   * @throws Exception
+   */
+  @Test
+  public void injectionAccountAndContainerForPutTest() throws Exception {
+    injectAccountAndContainerForPutAndVerify(true, shouldAllowServiceIdBasedPut);
+    injectAccountAndContainerForPutAndVerify(false, shouldAllowServiceIdBasedPut);
+  }
+
+  /**
+   * Tests injecting target {@link Account} and {@link Container} for GET/HEAD/DELETE requests. The {@link AccountService}
+   * is prepopulated with a reference account and {@link Account#UNKNOWN_ACCOUNT}. The expected behavior should be:
+   *
+   * AId in blobId    CId in blobId     expected Error      injected account      injected container
+   * realAId           realCId          NotFound            refAccount            refContainer
```

Comment:
Both `account` and `container` are valid, so the `account` and `container` checks are passed. However the blobId is hand-made and does not exist in the router, so it returns `NotFound`.